### DISCUSSION TITLE
Need: a unified way of making meshes

### DISCUSSION THREAD

**Discussion by adrifoster** [ESCOMP/CTSM#2811] [General]
We have realized that many of us (me, @slevis-lmwg, @swensosc, etc.) use different methods to create mesh files to run with the new NUOPC coupler. 

It would be good to figure out how those methods are different/similar and create a unified method.

Currently, we need a mesh creation script that can handle global, regional, and "sparse" grids for our use cases. The current mesh_maker script has some bugs associated with it, and I don't think would work for the sparse grid scenario.

Some issues we know of related to mesh maker/mesh creation:
#2774 
#2776 
#1731

**Comment by slevis-lmwg** [comment]
Linking an issue related to the "sparse" grid topic raised above:
#3099
